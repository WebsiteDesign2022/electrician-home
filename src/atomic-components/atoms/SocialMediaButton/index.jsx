import React from "react";

const SocialMediaButton = ({ name, link }) => (
    <a className="nav-link active" href={link}>{name}</a>
);

export default SocialMediaButton;
