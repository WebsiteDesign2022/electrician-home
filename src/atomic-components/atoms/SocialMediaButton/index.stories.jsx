import React from 'react';

import SocialMediaButton from '.';

export default {
  title: 'Electrician-Home/atoms/SocialMediaButton',
  component: SocialMediaButton,
  parameters: {
    // More on Story layout: https://storybook.js.org/docs/react/configure/story-layout
    layout: 'fullscreen',
  },
};

const Template = (args) => <SocialMediaButton {...args} />;

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Primary.args = {
  name: 'social-media-button-name',
  link: 'https://www.facebook.com'
};
