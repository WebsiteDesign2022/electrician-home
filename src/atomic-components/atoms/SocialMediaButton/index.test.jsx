import TestRenderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';

import SocialMediaButton from '.';

describe("SocialMediaButton", () => {
    test("renders SocialMediaButton", () => {
    
        render(<SocialMediaButton link="www.google.com" name="Google" />);

        expect(
            screen.getByRole("link", { link: 'www.google.com', name: 'Google' })
        ).toBeVisible();

        expect(
            screen.getByText(/Google/i)
        ).toBeInTheDocument();

    });

    test("renders SocialMediaButton", () => {
        const socialMediaButton = TestRenderer.create(
            <SocialMediaButton link="www.facebook.com" name="Facebook" />
        );

        expect(socialMediaButton.toJSON()).toMatchSnapshot();
    });
});
