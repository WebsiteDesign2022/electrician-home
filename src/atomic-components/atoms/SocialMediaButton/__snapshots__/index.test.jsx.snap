// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaButton renders SocialMediaButton 1`] = `
<a
  className="nav-link active"
  href="www.facebook.com"
>
  Facebook
</a>
`;
